source/includes/_node_rpc_mining.md

```diff
@@ -0,0 +1,618 @@
+});
+```
+
+> The above command returns JSON "result" like this:
+
+```json
+453742051556.55084
+```
+
+Returns the estimated current or historical network hashes per second, based on last `blocks`.
+
+### Params
+N. | Name | Default |  Description
+--------- | --------- | --------- | -----------
+1 | blocks | 120 | Number of blocks to lookup.
+2 | height | 1 | Starting height for calculations.
+
+
+
+## getmininginfo
+
+```shell--curl
+curl $url/ \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  --data '{
+    "method": "getmininginfo",
+    "params": []
+  }'
+```
+
+```shell--cli
+bcoin cli rpc getmininginfo
+```
+
+```javascript
+const rpc = new bcoin.http.RPCClient({
+  network: 'testnet'
+});
+
+(async () => {
+  const res = await rpc.execute('getmininginfo', []);
+
+  console.log(res);
+})().catch((err) => {
+  console.error(err.stack);
+});
+```
+
+> The above command returns JSON "result" like this:
+
+```json
+{
+  "blocks": 1178789,
+  "currentblocksize": 11895,
+  "currentblockweight": 47580,
+  "currentblocktx": 35,
+  "difficulty": 1048576,
+  "errors": "",
+  "genproclimit": 0,
+  "networkhashps": 10880012194348.812,
+  "pooledtx": 34,
+  "testnet": true,
+  "chain": "test",
+  "generate": false
+}
+
+```
+
+Returns mining info.
+
+*Note: currentblocksize, currentblockweight, currentblocktx, difficulty are returned when there's active work.*
+*generate - is true when `bcoin` itself is mining.*
+
+### Params
+N. | Name | Default |  Description
+--------- | --------- | --------- | -----------
+None. |
+
+
+
+## getwork
+
+```shell--curl
+curl $url/ \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  --data '{
+    "method": "getwork",
+    "params": []
+  }'
+```
+
+```shell--cli
+bcoin cli rpc getwork
+```
+
+```javascript
+const rpc = new bcoin.http.RPCClient({
+  network: 'testnet'
+});
+
+(async () => {
+  const res = await rpc.execute('getwork');
+
+  console.log(res);
+})().catch((err) => {
+  console.error(err.stack);
+});
+```
+
+> The above command returns JSON "result" like this:
+
+```json
+{
+  "data": "2000000082c18ff1d76f61dccbf37d5eca24d252c009af5f622ce08700000580000000009f447a03a1df9ae8f1ee24f0cf5ab69f1b321071f0e57344f2c61b922c12b8335994bdb51a0ffff000000000000000800000000000000000000000000000000000000000000000000000000000000000000000000000000080020000",
+  "target": "0000000000000000000000000000000000000000000000f0ff0f000000000000",
+  "height": 1178782
+}
+```
+
+Returns hashing work to be solved by miner.
+Or submits solved block.
+
+### Params
+N. | Name | Default |  Description
+--------- | --------- | --------- | -----------
+1 | data | | Data to be submitted to the network.
+
+## getworklp
+
+```shell--curl
+curl $url/ \
+  -H 'Content-Type: application/json' \
+  -X POST \
+  --data '{
+    "method": "getworklp",
+    "params": []
+  }'
+```
+
+```shell--cli
+# Because there's request timeout set on CLI http requests.
+# you need to send custom request with different timeout.
```

Comment:
Ok, so there's no way to change them even at a config level right now? "you need to send custom request" makes it sound like it is possible. If there's no way at all, maybe change it to something like: 
```
Because there is a request timeout set on CLI http requests
without manually adjusting the timeout, this call will timeout before the request is complete
```
Also, if the calls are still valid (even if they timeout) maybe we should still include it? That way if we expose a request timeout parameter this example will still work.